src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/taglets/TagletWriter.java

```diff
@@ -279,7 +279,8 @@ public Content getBlockTagOutput(TagletManager tagletManager,
                 }
             } catch (UnsupportedTagletOperationException e) {
                 // malformed taglet:
-                // claims to support block tags but does not provide the appropriate method.
+                // claims to support block tags (see Taglet.isBlockTag) but does not provide the
```

Comment:
This version does a better job of explaining your assumptions. Thanks.

(You don't need to address the below comment.)

I think this method could benefit from an assertion or an unconditional check that would guard against calling this method with a taglet list containing a non-block taglet. That is, a taglet `t` such that `t.isBlockTag() == false`.

This is a bit different from `TagletWriter#getInlineTagOutput` where the check is built into the method; `TagletWriter#getBlockTagOutput` relies on a caller to pass the correct list.

Reply:
Interesting question, worth considering ...

This is an internal API, so is only called from internal sites, with one of two lists supplied by the TagletManager: standard block tags, or serial form tags (the latter is why the list comes in as an argument and is not provided internally.)

But given the list comes in as an argument, that suggests a reasonable update is to verify that all items in the argument list are block tags, and to throw ILA if any are not.